### ISSUE TITLE
Creating cluster on ipv6 causes nodes to try to connect to port `0`

### ISSUE THREAD

**Issue by Defman** [microsoft/garnet#628]
### Describe the bug

When creating the bug the nodes are trying to connect to the master by using port 0...

```
    _________
   /_||___||_\      Garnet 1.0.19 64 bit; cluster mode
   '. \   / .'      Port: 6379
     '.\ /.'        https://aka.ms/GetGarnet
       '.'

* Ready to accept connections
07::28::44 fail: ClusterManager[0] Meet terminated with error System.Net.Sockets.SocketException (113): No route to host [2a05:d014:1915:aa02:f86d:0:7:6379]:0    at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)    at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)    at Garnet.client.GarnetClient.GetSendSocket(Int32 millisecondsTimeout) in /src/libs/client/GarnetClient.cs:line 274    at Garnet.client.GarnetClient.ConnectAsync(CancellationToken token) in /src/libs/client/GarnetClient.cs:line 220    at Garnet.client.GarnetClient.ReconnectAsync(CancellationToken token) in /src/libs/client/GarnetClient.cs:line 343    at Garnet.cluster.GarnetServerNode.Initialize() in /src/libs/cluster/Server/GarnetServerNode.cs:line 97    at Garnet.cluster.ClusterManager.Meet(String address, Int32 port) in /src/libs/cluster/Server/Gossip.cs:line 266
07::32::50 fail: ClusterManager[0] Meet terminated with error System.Net.Sockets.SocketException (113): No route to host [2a05:d014:1915:aa02:f86d:0:7:6379]:0    at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)    at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)    at Garnet.client.GarnetClient.GetSendSocket(Int32 millisecondsTimeout) in /src/libs/client/GarnetClient.cs:line 274    at Garnet.client.GarnetClient.ConnectAsync(CancellationToken token) in /src/libs/client/GarnetClient.cs:line 220    at Garnet.client.GarnetClient.ReconnectAsync(CancellationToken token) in /src/libs/client/GarnetClient.cs:line 343    at Garnet.cluster.GarnetServerNode.Initialize() in /src/libs/cluster/Server/GarnetServerNode.cs:line 97    at Garnet.cluster.ClusterManager.Meet(String address, Int32 port) in /src/libs/cluster/Server/Gossip.cs:line 266
```

### Steps to reproduce the bug

1. create three nodes bound to address `[::]`(ipv6)
2. 
```
redis-cli --cluster create \
    garent-0.garent-headless.default.svc.cluster.local:6379 \
    garent-1.garent-headless.default.svc.cluster.local:6379 \
    garent-2.garent-headless.default.svc.cluster.local:6379 \
    --cluster-yes
```
or
```
redis-cli --cluster create \
    [2a05:d014:1915:aa02:f86d::7]:6379 \
    [2a05:d014:1915:aa02:f86d::b]:6379 \
    [2a05:d014:1915:aa02:f86d::c]:6379 \
    --cluster-yes
```
or
```
redis-cli --cluster create \
    2a05:d014:1915:aa02:f86d::7:6379 \
    2a05:d014:1915:aa02:f86d::b:6379 \
    2a05:d014:1915:aa02:f86d::c:6379 \
    --cluster-yes
```
or
```
redis-cli --cluster create \
    [2a05:d014:1915:aa02:f86d:0:0:7]:6379 \
    [2a05:d014:1915:aa02:f86d:0:0:b]:6379 \
    [2a05:d014:1915:aa02:f86d:0:0:c]:6379 \
    --cluster-yes
```

### Expected behavior

Use the given port instead of defaulting to 0

**Comment by vazois** [comment]
Thanks for the detailed repro. I have a fix here #636. Can you check and let me know if that works for you know?

**Comment by Defman** [comment]
Yes, this seems to have solved the issue.
```
>>> Performing hash slots allocation on 3 nodes...
Master[0] -> Slots 0 - 5460
Master[1] -> Slots 5461 - 10922
Master[2] -> Slots 10923 - 16383

>>> Nodes configuration updated
>>> Assign a different config epoch to each node
>>> Sending CLUSTER MEET messages to join the cluster
Waiting for the cluster to join
..

[OK] All nodes agree about slots configuration.
>>> Check for open slots...
>>> Check slots coverage...
[OK] All 16384 slots covered
```
